kdb/keyset.go

```diff
@@ -25,8 +26,11 @@ type KeySet interface {
 	Pop() Key
 	Head() Key
 	Tail() Key
+	// TODO REVIEW API: We should remove the internal iterator and provide an external instead
 	Next() Key
 	Len() int
+
+	// TODO REVIEW API: We should remove the internal iterator and provide an external instead
```

Comment:
What is the difference between an internal and external iterator? Do you mean Next() and Rewind should not be exposed - but abstracted?

Reply:
Next and Rewind is the internal iterator. It should not be used as we want to get rid of it, see #2991. Instead cursor_t and ksAtCursor should be used. This should be nicely abstracted in Go.